### ISSUE TITLE
overlayfs snapshot driver

### ISSUE THREAD

**Issue by jmccann** [vito/oci-build-task#50]
I'm not sure for others but for me when using this image to build other images it's not using overlayfs as the snapshot driver but rather the naive driver.  This means when unpacking an image each layer unpacked is a copy of previous layer with addition of new content rather then having an overlay of just the difference.  This is time consuming with the disk IO involved and can use a lot more disk space to build images.

When I intercept a step using this image I see

```
ps -ef | grep buildkitd
130 root      2:21 buildkitd --root /tmp/buildkitd --addr unix:///tmp/buildkitd/buildkitd.sock
```

If I wait for the build to finish and start buildkitd myself adding `--debug` I see

```
~ # buildkitd --root /tmp/buildkitd --addr unix:///tmp/buildkitd/buildkitd.sock --debug
DEBU[0000] auto snapshotter: overlayfs is not available for /tmp/buildkitd, trying fuse-overlayfs: failed to mount overlay: invalid argument 
```

If I change `--root` to use a filesystem that isn't on overlay it is then able to use overlayfs driver

```
~ # df -h /tmp
Filesystem                Size      Used Available Use% Mounted on
overlay                  49.0G     12.2G     34.2G  26% /
~ # df -h /scratch
Filesystem                Size      Used Available Use% Mounted on
/dev/vdb                 49.0G     12.2G     34.2G  26% /scratch
~ # buildkitd --root /scratch/tmp/buildkitd --addr unix:///tmp/buildkitd/buildkitd.sock --debug
INFO[0000] auto snapshotter: using overlayfs            
...
```

Not sure if we can just trust `/scratch` to exist but I think we should figure out using a `--root` that will take advantage of overlayfs snapshot driver.

**Comment by vito** [comment]
Ah yeah dang not sure how I didn't notice that. Now I wonder if it's been using naive in our pipelines this whole time. 😱  `/scratch` exists for exactly this purpose. I'll take a look shortly! Thanks for submitting #51, though I'll probably just hardcode the path and close it.

**Comment by vito** [comment]
Fixed in v0.5.0, just released. Thanks again!
